### DISCUSSION TITLE
Honor 7x

### DISCUSSION THREAD

**Discussion by Viking8** [phhusson/treble_experimentations#1888] [Q&A]
Hi,

I was given an Honor 7x recently. I've seen sites that say it can use the ROM's here on this site as an upgrade. My question... Is it recommended to do the upgrade to this ROM while still on the stock Oreo bootloader, or should I upgrade to EMUI 9.1 before unlocking the Bootloader with PotatoNV? Also, should I stick with Android 10 or will the Android 11 ROM's here be compatible?

Thanks!

**Comment by Viking8** [comment]
Anyone?

**Comment by CN-Mark-Zhang** [comment]
You'd better upgrade to EMUI 9.1 first because it's the only way to install Android 12 and further updates.
